src/edge.ts

```diff
@@ -1,15 +1,9 @@
 export default class Edge {
-  direction: string;
+  direction: 'horizontal' | 'vertical';
   position: number;
   type?: string
 
-  constructor(
-    options: {
-      direction: string,
-      position: number,
-      type?: string
-    }
-  ) {
+  constructor(options: { direction: 'horizontal' | 'vertical', position: number, type?: string }) {
```

Comment:
I've seen this done as a Type before ... not sure if it would benefit from it though 🤔 I like the idea of it being a type so it's a little more concise and less repetition but can also only see this being use in two places so happy to have this here as it is... Team to decide 👯‍♂️

Reply:
As it's used in class and constructor, a type might make sense but in the same file unless it's used elsewhere. Something like `DirectionType`?

Reply:
Like this?
```ts
type DirectionType = {
  direction: 'horizontal' | 'vertical';
  position: number;
  type?: string;
};

export default class Edge {
  direction: 'horizontal' | 'vertical';
  position: number;
  type?: string;

  constructor(options: DirectionType) {
    this.direction = options.direction;
    this.position = options.position;
    this.type = options.type;
  }
}
```